functions.inc.php

```diff
@@ -1021,6 +1047,64 @@ function pacrypt ($pw, $pw_db="") {
     return $password;
 }
 
+// used for php_crypt method
+function random_string($characters, $length) {
+    $string = '';
+    for ($p = 0; $p < $length; $p++) {
+        // should really use a stronger randomness source
+        $string .= $characters[mt_rand(0, strlen($characters)-1)];
```

Comment:
mt_rand is NOT cryptographically secure and should not be used for salt generation. Check out random_bytes /  openssl_random_pseudo_bytes /  mcrypt_create_iv functions.